velox/expression/ExprCompiler.h

```diff
@@ -31,4 +31,12 @@ std::vector<std::shared_ptr<Expr>> compileExpressions(
     ExprSet* exprSet,
     bool enableConstantFolding = true);
 
+struct ITypedExprComparer {
+  bool operator()(const core::ITypedExpr* lhs, const core::ITypedExpr* rhs)
+      const {
+    // Add check for type as == operator only checks for type kind equivalence
+    return (*lhs == *rhs) && (lhs->type() == rhs->type());
```

Comment:
can this have any unintended results? we currently seem to check for weak equivalence in typedExpressions for reasons listed here (https://github.com/facebookincubator/velox/blob/main/velox/type/Type.h#L489-L493) but operator== would check for strong equivalence. With a change like this, can we run in a situation where 2 expressions are wrongly categorized as inequivalent?